src/Type/UnionType.php

```diff
@@ -102,22 +102,62 @@ private function getSortedTypes(): array
 	 */
 	public function getReferencedClasses(): array
 	{
-		return UnionTypeHelper::getReferencedClasses($this->getTypes());
+		$classes = [];
+		foreach ($this->types as $type) {
+			foreach ($type->getReferencedClasses() as $className) {
+				$classes[] = $className;
+			}
+		}
+
+		return $classes;
 	}
 
 	public function getArrays(): array
 	{
-		return UnionTypeHelper::getArrays($this->getTypes());
+		$arrays = [];
+		foreach ($this->types as $type) {
+			foreach ($type->getArrays() as $array) {
+				$arrays[] = $array;
```

Comment:
I think getArrays should behave the same way as getConstantArrays here.

Reply:
implemented it like https://github.com/phpstan/phpstan-src/blob/5a0d7368bf3540a58626c28708c45c933fe2dda2/src/Type/TypeUtils.php#L20-L64

not sure what todo with the generalization test error